analyze.py

```diff
@@ -382,6 +382,36 @@ def analyze_timeline(user_id, list_id, api, cache):
     users = fetch_users(timeline_ids, api, cache)
     return analyze_users(users, ids_fetched=len(timeline_ids))
 
+def analyze_my_timeline(user_id, api, cache):
+    # Timeline-functions are limited to 200 statuses
+    statuses = api.GetUserTimeline(screen_name=user_id, count=200, include_rts=True, trim_user=False, exclude_replies=False)
+    maxt = 0
+    for i in range(1, 10): #end of range is exclusive so this only goes to 9, so we're looking at max 2,000 tweets
+        if not maxt == statuses[-1].id - 1: #to prevent excessive API calls if already exhausted
+            maxt = statuses[-1].id - 1
+            statuses = statuses + api.GetUserTimeline(screen_name=user_id, count=200, max_id=maxt, include_rts=True, trim_user=False, exclude_replies=False)    
```

Comment:
I'm surprised that sending statuses[-1].id as "max_id" does not result in that tweet appearing twice in the `statuses` array, but it does seem to work somehow. Perhaps Twitter's API docs are wrong when [they say this method returns statuses less than or equal to max_id](https://developer.twitter.com/en/docs/tweets/timelines/api-reference/get-statuses-user_timeline); maybe it's strictly less than.

Reply:
Thanks - I think it works because of the minus 1; I think if max_id were defined as just statuses[-1].id you're right that there would be duplication.